File: calculator-dotnet/Models/CalculationResult.cs
namespace CalculatorApp.Models;

/// <summary>Represents the result of a calculation.</summary>
/// <param name="FirstNumber">The first operand used.</param>
/// <param name="SecondNumber">The second operand used.</param>
/// <param name="Operation">The operation that was performed.</param>
/// <param name="Symbol">The display symbol for the operation.</param>
/// <param name="Result">The computed result.</param>
public sealed record CalculationResult(
    double FirstNumber,
    double SecondNumber,
    Operation Operation,
    string Symbol,
    double Result
);

File: src/main/java/com/example/calculator/model/CalculationResult.java
package com.example.calculator.model;

/**
 * Immutable record that represents the result of a single calculation.
 *
 * <p>Mirrors the "computation details" expanded panel from the original Python
 * Streamlit app.
 *
 * @param firstNumber  the left-hand operand used in the calculation
 * @param secondNumber the right-hand operand used in the calculation
 * @param operation    the operation that was applied
 * @param result       the numeric result
 * @param expression   a human-readable string such as {@code "3.0 + 4.0 = 7.0"}
 */
public record CalculationResult(
        double firstNumber,
        double secondNumber,
        Operation operation,
        double result,
        String expression) {

    /**
     * Convenience factory that builds a {@code CalculationResult} and
     * pre-formats the {@code expression} field.
     */
    public static CalculationResult of(double a, double b, Operation op, double result) {
        String expression = "%s %s %s = %s".formatted(a, op.symbol(), b, result);
        return new CalculationResult(a, b, op, result, expression);
    }
}
